src/dirtyfields/dirtyfields.py

```diff
@@ -27,12 +31,14 @@ def __init__(self, *args, **kwargs):
                 name=self.__class__.__name__))
         if self.ENABLE_M2M_CHECK:
             self._connect_m2m_relations()
-        reset_state(sender=self.__class__, instance=self)
+        if not self.RESET_STATE_ON_SAVE_ONLY:
+            reset_state(sender=self.__class__, instance=self)
```

Comment:
if `RESET_STATE_ON_SAVE_ONLY` is True, the original state will never get set then?

Reply:
Yeah, what's the intention of putting this init stuff behind the flag?

Reply:
We still need to call `reset_state`, otherwise non-m2m changes are not tracked.